src/apps/api/serializers/leaderboards.py

```diff
@@ -105,3 +107,52 @@ def get_submissions(self, instance):
 
         submissions = submissions.order_by(*ordering, 'created_when')
         return SubmissionLeaderBoardSerializer(submissions, many=True).data
+
+
+class LeaderboardPhaseSerializer(serializers.ModelSerializer):
+    submissions = serializers.SerializerMethodField(read_only=True)
+    tasks = TaskSerializer(many=True, read_only=True)
+    columns = serializers.SerializerMethodField()
+
+    def get_columns(self, instance):
+        columns = Column.objects.filter(leaderboard=instance.leaderboard)
+        if len(columns) == 0:
+            raise serializers.ValidationError("No columns exist on the leaderboard")
+        elif len(columns) == 1:
+            return ColumnSerializer(columns).data
+        else:
+            return ColumnSerializer(columns, many=True).data
+
+    class Meta:
+        model = Phase
+        fields = (
+            'id',
+            'name',
+            'submissions',
+            'tasks',
+            'leaderboard',
+            'columns',
+        )
+        depth = 1
+
+    def get_submissions(self, instance):
+        # desc == -colname
+        # asc == colname
+        primary_col = instance.leaderboard.columns.get(index=instance.leaderboard.primary_index)
+        ordering = [f'{"-" if primary_col.sorting == "desc" else ""}primary_col']
+        submissions = Submission.objects.filter(
```

Comment:
is this a duplication of the above query I commented on? looks similar, may be slightly different?

Reply:
I think what this code does is first get the primary column's order. Then goes through all columns excluding the primary column. My guess is this code orders by the primary column, then uses other columns as tiebreakers? @dde6khkg